lib/sync-server.js

```diff
@@ -105,7 +105,16 @@ var DEFAULT_SYNC_CONF = {
   /** @type {Boolean} Specify if the server should wait for the ack insert to complete before returning the response for the sync request. Default is true. */
   syncReqWaitForAck: true,
   /** @type {Number} Specify the max number of ack items will be processed for a single request. Default is -1 (unlimited).*/
-  syncReqAckLimit: -1
+  syncReqAckLimit: -1,
+  /** @type {Function} Provide your own cuid generator. It should be a function and the `params` object will be passed to the generator. 
+   *  The `params` object will have the following fields:
+   *  `params.query_params`: the query params used on the dataset
+   *  `params.meta_data`: the meta data used on the dataset
+   *  `params.__fh.cuid`: the cuid generated on the client.
+   * 
+   *  This function should not be provided in most of the cases. This should *ONLY* be provided if there is a chance that the clients may have duplicated cuids.
```

Comment:
We can update this comment when we make that change. For now I just want to discourage people from using it, unless they are absolute sure about it.